src/Socket.php

```diff
@@ -103,94 +100,104 @@ public function defineTabs($options = [])
         return $ong;
     }
 
-
     /**
      * NetworkPort Form.
      *
      * @return void
+     * @used-by templates/pages/assets/socket.html.twig
      */
     public static function showNetworkPortForm($itemtype, $items_id, $networkports_id = 0, $options = [])
     {
-
-        /** @var array $CFG_GLPI */
-        global $CFG_GLPI;
-
-        //if form is called from an item, retrieve itemtype and items
+        // if form is called from an item, retrieve itemtype and items
         if (isset($options['_add_fromitem'])) {
             $itemtype = $options['_add_fromitem']["_from_itemtype"];
             $items_id = $options['_add_fromitem']["_from_items_id"];
         }
 
-        $rand_itemtype = rand();
-        $rand_items_id = rand();
-
-        echo "<span id='show_itemtype_field' class='input_listener'>";
-        Dropdown::showFromArray('itemtype', self::getSocketLinkTypes(), ['value' => $itemtype,
-            'rand' => $rand_itemtype
-        ]);
-        echo "</span>";
-
-        $params = ['itemtype'   => '__VALUE__',
-            'dom_rand'   => $rand_items_id,
-            'dom_name'   => 'items_id',
-            'action'     => 'get_items_from_itemtype'
+        $rand_itemtype = mt_rand();
+        $rand_items_id = mt_rand();
+
+        $twig_params = [
+            'several' => $options['several'] ?? false,
+            'socket_link_types' => self::getSocketLinkTypes(),
+            'itemtype' => $itemtype,
+            'items_id' => $items_id,
+            'networkports_id' => $networkports_id,
+            'rand_itemtype' => $rand_itemtype,
+            'rand_items_id' => $rand_items_id,
         ];
-        Ajax::updateItemOnSelectEvent(
-            "dropdown_itemtype$rand_itemtype",
-            "show_items_id_field",
-            $CFG_GLPI["root_doc"] . "/ajax/cable.php",
-            $params
-        );
-
-        echo "<span id='show_items_id_field' class='input_listener'>";
-        if (!empty($itemtype)) {
-            $rand_items_id =  $itemtype::dropdown(['name'                  => 'items_id',
-                'value'                 => $items_id,
-                'display_emptychoice'   => true,
-                'display_dc_position'   => in_array($itemtype, $CFG_GLPI['rackable_types']),
-                'rand' => $rand_items_id
-            ]);
-        }
-        echo "</span>";
-
+        echo TemplateRenderer::getInstance()->renderFromStringTemplate(<<<TWIG
```

Comment:
`TemplateRenderer::renderFromStringTemplate()` should be used only for really small templates. Indeed, templates used this way:
 - are not parsed by `xgettext` and their translations are therefore not recognized,
 - are not parsed by linters.